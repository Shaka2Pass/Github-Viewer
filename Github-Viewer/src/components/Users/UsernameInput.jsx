import React from 'react';
import PropTypes from 'prop-types';

const UsernameInput = ({ username, onNameChange }) => (
  <>
    <input type="text" value={username} onChange={onNameChange} />
  </>
);

UsernameInput.propTypes = {
  username: PropTypes.string.isRequired,
  onNameChange: PropTypes.func.isRequired
};

export default UsernameInput;

