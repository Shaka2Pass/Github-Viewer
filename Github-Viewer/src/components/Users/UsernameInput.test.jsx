import React from 'react';
import { shallow } from 'enzyme';
import UsernameInput from './UsernameInput';

describe('User', () => {
  it('matches a user', () => {
    const wrapper = shallow(<UsernameInput username="shaka2pass" />);
    expect(wrapper).toMatchSnapshot();
  });
});

