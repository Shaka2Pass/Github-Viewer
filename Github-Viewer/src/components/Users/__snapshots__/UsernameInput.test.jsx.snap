// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a user 1`] = `
<Fragment>
  <input
    type="text"
    value="shaka2pass"
  />
</Fragment>
`;
